File: SPCode/Modules/CultureSupport.vb

Imports System.Text.RegularExpressions


Module CultureSupport

    Public Enum TextReturnTypeEnum
        PureString
        RawString
    End Enum

    Private m_dctLanguageDictionary As Hashtable

    Public Property LanguageDictionary() As System.Collections.Hashtable
        Get
            Return m_dctLanguageDictionary
        End Get
        Set(ByVal Value As System.Collections.Hashtable)
            m_dctLanguageDictionary = Value
        End Set
    End Property
    Public Function TextDictionary(ByVal TextID As String, Optional ByRef ParseType As TextReturnTypeEnum = TextReturnTypeEnum.PureString) As String
        Try
            Select Case ParseType
                Case TextReturnTypeEnum.PureString
                    Return DeleteChar(CStr(LanguageDictionary(TextID.Trim)))
                Case TextReturnTypeEnum.RawString
                    Return CStr(LanguageDictionary(TextID.Trim))
            End Select
        Catch ex As Exception

        End Try

        Return vbNullString
    End Function
End Module

Module DateSupport
    Public Function DateCheck(ByRef strValue As String) As Boolean
        DateCheck = True ' Assume True
        If strValue <> "" Then
            If Not IsDate(strValue) Then
                Return False
            End If
        End If
    End Function

End Module

Module StringSupport
    Public Function Substitute(ByRef sString As String, ByRef sFind As String, ByRef sReplace As String) As String
        ' Substitutes string sReplace in place of string sFind in sString
        Dim lEnd, lStart, lFindLength As Integer
        Dim sNewString As String

1:      sNewString = ""
2:      lFindLength = Len(sFind)
3:      lStart = 1
4:      lEnd = InStr(lStart, sString, sFind)
5:      Do While lEnd > 0
6:          sNewString = sNewString & Mid(sString, lStart, lEnd - lStart) & sReplace
7:          lStart = lEnd + lFindLength
8:          lEnd = InStr(lStart, sString, sFind)
9:      Loop
10:     Substitute = sNewString & Mid(sString, lStart)
    End Function
    Public Function RemoveAmpersands(ByRef strString As String) As String
        ' Removes the ampersands in the specified string.
1:      RemoveAmpersands = Substitute(strString, "&", "")
    End Function
    Public Function NullTruncate(ByVal strText As String) As String
        ' Returns the specified string truncated at the first null character
        Dim lLen As Integer

1:      lLen = InStr(strText, Chr(0))
2:      If lLen < 1 Then
3:          Return strText
4:      Else
5:          Return Left(strText, lLen - 1)
6:      End If
    End Function
    Public Function ResolveResString(ByVal resID As Short, ByVal ParamArray varReplacements() As Object) As String
        '-----------------------------------------------------------
        ' FUNCTION: ResolveResString
        ' Reads resource and replaces given macros with given values
        '
        ' Example, given a resource number 14:
        '    "Could not read '|1' in drive |2"
        '   The call
        '     ResolveResString(14, "|1", "TXTFILE.TXT", "|2", "A:")
        '   would return the string
        '     "Could not read 'TXTFILE.TXT' in drive A:"
        '
        ' IN: [resID] - resource identifier
        '     [varReplacements] - pairs of macro/replacement value
        '-----------------------------------------------------------
        '
        Dim intMacro As Integer
        Dim strResString As String = vbNullString

1:      'strResString = VB6.LoadResString(resID)

        ' For each macro/value pair passed in...
2:      Dim strMacro As String
        Dim strValue As String
        Dim intPos As Integer
        For intMacro = 0 To varReplacements.Length Step 2

3:          'UPGRADE_WARNING: Couldn't resolve default property of object varReplacements(). Click for more: 'ms-help://MS.VSCC.2003/commoner/redir/redirect.htm?keyword="vbup1037"'
            strMacro = varReplacements(intMacro).ToString
4:          On Error GoTo MismatchedPairs
5:          'UPGRADE_WARNING: Couldn't resolve default property of object varReplacements(). Click for more: 'ms-help://MS.VSCC.2003/commoner/redir/redirect.htm?keyword="vbup1037"'
            strValue = varReplacements(intMacro + 1).ToString
6:          On Error GoTo 0

            ' Replace all occurrences of strMacro with strValue
7:          Do
8:              intPos = InStr(strResString, strMacro)
9:              If intPos > 0 Then
10:                 strResString = Left(strResString, intPos - 1) & strValue & Right(strResString, Len(strResString) - Len(strMacro) - intPos + 1)
11:             End If
12:         Loop Until intPos = 0
13:     Next intMacro

14:     ResolveResString = strResString

15:     Exit Function

MismatchedPairs:
16:     Resume Next
    End Function
    Public Function DeleteChar(ByVal SourceString As String, Optional ByVal SourceCharacters As String = "&,:") As String
        '-------------------------------------------------------------------------------
        '
        ' DeleteChar - Function to delete a specified char in a passed string
        '              The characters might be separated by a comma ','
        '
        ' Parameters:
        '    srField2String - I:   (String) holds the string to be processed
        '    srcChar   - I:   (String)holds the characters to remove from string
        '
        ' Returns:
        '   String
        '
        ' Comments:
        '   None.
        '
        '-------------------------------------------------------------------------------

        Const PkComma As String = ","

        Dim astrSource() As String
        Dim finalString As String
        Dim asrcSplitChar() As String

        If SourceString = vbNullString Then
            DeleteChar = vbNullString
            Exit Function
        End If

        'Extract the characters to delete from source string
        asrcSplitChar = Split(SourceCharacters, PkComma, , CompareMethod.Binary)

        'Exit if no characters to delete
        If asrcSplitChar.Length = 0 Then
            DeleteChar = SourceString
            Exit Function
        End If

        finalString = SourceString
        'Loop through the characters and delete each from source string
        For Each strChar As String In asrcSplitChar
            astrSource = Split(finalString, strChar, , CompareMethod.Binary)
            finalString = Join(astrSource, vbNullString)
        Next


        DeleteChar = finalString
    End Function

    Public Function ValidateEmail(ByVal Expression As String) As Boolean
        Dim arrstrEmail() As String
        Dim idOk As Boolean

        Try
            arrstrEmail = Split(Expression, "@")
            If arrstrEmail.Length > 0 Then
                idOk = (arrstrEmail(arrstrEmail.Length) Like "*.*") And _
                (arrstrEmail(0) <> vbNullString) And _
                (Len(arrstrEmail(UBound(arrstrEmail))) > 3)
            End If
        Catch ex As Exception

        End Try
        Return idOk
    End Function

    Public Function IsAlphabet(ByVal strExp As String) As Boolean
        Return Regex.IsMatch(strExp, "^[a-zA-Z]+$")
    End Function

    Public Function IsValidEmail(ByVal strExp As String) As Boolean
        Return Regex.IsMatch(strExp, "^[a-zA-Z][\w\.-]*[a-zA-Z0-9]@[a-zA-Z][\w\.-]*[a-zA-Z0-9]\.[a-zA-Z][a-zA-Z\.]*[a-zA-Z]$")
    End Function

    Public Function IsValidTime(ByVal strExp As String) As Boolean
        Return Regex.IsMatch(strExp, "(([01]+[\d]+)/(2[0-3])):[0-5]+[0-9]+")
    End Function

End Module

File: SPCode/CS/Modules/CultureSupport.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;
using Microsoft.VisualBasic.CompilerServices;




namespace SoftLogik.Win
{
	sealed class CultureSupport
	{
		
		public enum TextReturnTypeEnum
		{
			PureString,
			RawString
		}
		
		private static Hashtable m_dctLanguageDictionary;
		
		public static System.Collections.Hashtable LanguageDictionary
		{
			get
			{
				return m_dctLanguageDictionary;
			}
			set
			{
				m_dctLanguageDictionary = value;
			}
		}
		public static string TextDictionary(string TextID, TextReturnTypeEnum ParseType)
		{
			try
			{
				switch (ParseType)
				{
					case TextReturnTypeEnum.PureString:
						return StringSupport.DeleteChar(LanguageDictionary[TextID.Trim()].ToString(), "&,:");
					case TextReturnTypeEnum.RawString:
						return LanguageDictionary[TextID.Trim()].ToString();
				}
			}
			catch (Exception)
			{
				
			}
			
			return Constants.vbNullString;
		}
	}
	
	sealed class DateSupport
	{
		public static bool DateCheck(string strValue)
		{
			bool returnValue;
			returnValue = true; // Assume True
			if (strValue != "")
			{
				if (! Information.IsDate(strValue))
				{
					return false;
				}
			}
			return returnValue;
		}
		
	}
	
	sealed class StringSupport
	{
		public static string Substitute(string sString, string sFind, string sReplace)
		{
			string returnValue;
			// Substitutes string sReplace in place of string sFind in sString
			int lEnd;
			int lStart;
			int lFindLength;
			string sNewString;
			
_1:
			sNewString = "";
_2:
			lFindLength = sFind.Length;
_3:
			lStart = 1;
_4:
			lEnd = sString.IndexOf(sFind, lStart - 1) + 1;
_5:
			while (lEnd > 0)
			{
_6:
				sNewString = sNewString + sString.Substring(lStart - 1, lEnd - lStart) + sReplace;
_7:
				lStart = lEnd + lFindLength;
_8:
				lEnd = sString.IndexOf(sFind, lStart - 1) + 1;
_9:
				1.GetHashCode() ; //nop
			}
_10:
			returnValue = sNewString + sString.Substring(lStart - 1);
			return returnValue;
		}
		public static string RemoveAmpersands(ref string strString)
		{
			// Removes the ampersands in the specified string.
1: RemoveAmpersands = Substitute(strString, "&", "");
		}
		public static string NullTruncate(string strText)
		{
			// Returns the specified string truncated at the first null character
			int lLen;
			
_1:
			lLen = strText.IndexOf(('\0').ToString()) + 1;
_2:
			if (lLen < 1)
			{
_3:
				return strText;
_4:
			}
			else
			{
_5:
				return strText.Substring(0, lLen - 1);
_6:
				1.GetHashCode() ; //nop
			}
		}
		public static string ResolveResString(short resID, params object[] varReplacements)
		{
			string returnValue;
			//-----------------------------------------------------------
			// FUNCTION: ResolveResString
			// Reads resource and replaces given macros with given values
			//
			// Example, given a resource number 14:
			//    "Could not read '|1' in drive |2"
			//   The call
			//     ResolveResString(14, "|1", "TXTFILE.TXT", "|2", "A:")
			//   would return the string
			//     "Could not read 'TXTFILE.TXT' in drive A:"
			//
			// IN: [resID] - resource identifier
			//     [varReplacements] - pairs of macro/replacement value
			//-----------------------------------------------------------
			//
			int intMacro;
			string strResString = Constants.vbNullString;
			
_1: //strResString = VB6.LoadResString(resID)

			// For each macro/value pair passed in...
_2:
			string strMacro;
			string strValue;
			int intPos;
			for (intMacro = 0; intMacro <= varReplacements.Length; intMacro += 2)
			{
				
_3: //UPGRADE_WARNING: Couldn't resolve default property of object varReplacements(). Click for more: 'ms-help://MS.VSCC.2003/commoner/redir/redirect.htm?keyword="vbup1037"'
				strMacro = varReplacements[intMacro].ToString();
_4:
				//On Error Goto MismatchedPairs VBConversions Warning: On Error Goto not supported in C#
_5: //UPGRADE_WARNING: Couldn't resolve default property of object varReplacements(). Click for more: 'ms-help://MS.VSCC.2003/commoner/redir/redirect.htm?keyword="vbup1037"'
				strValue = varReplacements[intMacro + 1].ToString();
_6:
				Microsoft.VisualBasic.CompilerServices.ProjectData.ClearProjectError();
				
				// Replace all occurrences of strMacro with strValue
_7:
				do
				{
_8:
					intPos = strResString.IndexOf(strMacro) + 1;
_9:
					if (intPos > 0)
					{
_10:
						strResString = strResString.Substring(0, intPos - 1) + strValue + strResString.Substring(strResString.Length - strResString.Length - strMacro.Length - intPos + 1, strResString.Length - strMacro.Length - intPos + 1);
_11:
						1.GetHashCode() ; //nop
					}
_12:
					1.GetHashCode() ; //nop
				} while (!(intPos == 0));
_13:
				1.GetHashCode() ; //nop
			}
			
_14:
			returnValue = strResString;
			
_15:
			return returnValue;
			
MismatchedPairs:
_16:
			1.GetHashCode() ; //nop
			//On Error Resume Next VBConversions Warning: On Error Resume Next not supported in C#
			return returnValue;
		}
		public static string DeleteChar(string SourceString, string SourceCharacters)
		{
			string returnValue;
			//-------------------------------------------------------------------------------
			//
			// DeleteChar - Function to delete a specified char in a passed string
			//              The characters might be separated by a comma ','
			//
			// Parameters:
			//    srField2String - I:   (String) holds the string to be processed
			//    srcChar   - I:   (String)holds the characters to remove from string
			//
			// Returns:
			//   String
			//
			// Comments:
			//   None.
			//
			//-------------------------------------------------------------------------------
			
			const string PkComma = ",";
			
			string[] astrSource;
			string finalString;
			string[] asrcSplitChar;
			
			if (SourceString == Constants.vbNullString)
			{
				returnValue = Constants.vbNullString;
				return returnValue;
			}
			
			//Extract the characters to delete from source string
			asrcSplitChar = Strings.Split(SourceCharacters, PkComma, -1, CompareMethod.Binary);
			
			//Exit if no characters to delete
			if (asrcSplitChar.Length == 0)
			{
				returnValue = SourceString;
				return returnValue;
			}
			
			finalString = SourceString;
			//Loop through the characters and delete each from source string
			foreach (string strChar in asrcSplitChar)
			{
				astrSource = Strings.Split(finalString, strChar, -1, CompareMethod.Binary);
				finalString = string.Join(Constants.vbNullString, astrSource);
			}
			
			
			returnValue = finalString;
			return returnValue;
		}
		
		public static bool ValidateEmail(string Expression)
		{
			string[] arrstrEmail;
			bool idOk;
			
			try
			{
				arrstrEmail = Expression.Split('@');
				if (arrstrEmail.Length > 0)
				{
					idOk = (StringType.StrLike(arrstrEmail[arrstrEmail.Length], "*.*", CompareMethod.Binary)) && (arrstrEmail[0] != Constants.vbNullString) && (arrstrEmail[(arrstrEmail.Length - 1)].Length > 3);
				}
			}
			catch (Exception)
			{
				
			}
			return idOk;
		}
		
		public static bool IsAlphabet(string strExp)
		{
			return Regex.IsMatch(strExp, "^[a-zA-Z]+$");
		}
		
		public static bool IsValidEmail(string strExp)
		{
			return Regex.IsMatch(strExp, "^[a-zA-Z][\\w\\.-]*[a-zA-Z0-9]@[a-zA-Z][\\w\\.-]*[a-zA-Z0-9]\\.[a-zA-Z][a-zA-Z\\.]*[a-zA-Z]$");
		}
		
		public static bool IsValidTime(string strExp)
		{
			return Regex.IsMatch(strExp, "(([01]+[\\d]+)/(2[0-3])):[0-5]+[0-9]+");
		}
		
	}
	
	
}
